stout/buffer.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <string>
+#include <utility> // std::move
+
+#include "uv.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+// Moveable and Copyable.
+class Buffer {
```

Comment:
Let's actually put this into `EventLoop`, similar to how `Clock` is in there. It does make `EventLoop` a "big" class but we can pull out function implementations later if we need/want. I don't want to pollute the eventuals namespace with `Buffer` and this `Buffer` is really an `EventLoop::Buffer`.